doc/introduction/photonic_hardware.rst

```diff
@@ -46,39 +46,23 @@ will execute.
 
 Jobs can also be submitted using the **non-blocking** :meth:`eng.run_async() <.RemoteEngine.run_async>`
 method. Unlike :meth:`eng.run() <.RemoteEngine.run>`, which returns a :class:`~.Results` object once the computation is
-complete, this method instead returns a :class:`~.Job` object directly that can be queried
+complete, this method instead returns a `xcc.Job` object directly that can be queried
 to get the job's status.
 
 >>> job = engine.run_async(prog, shots=3)
 >>> job.status
 "queued"
->>> job.refresh()
+>>> job.clear()
 >>> job.status
 "complete"
->>> result = job.result
+>>> result = sf.api.Result(job.result)
```

Comment:
minor question: should `Result` be part of the XCC rather than SF?

Reply:
Nope. As far as the XCC is concerned, the result of a job is a dictionary (which is parsed from an .npz file); it has no concept of "samples" or the like.

Reply:
:+1: I suppose it's just a matter of migrating `sf.api.Result` out of the API folder?

Reply:
Hmm, I don't quite follow. 🤔 Can you clarify what is achieved by moving sf.api.Result out of the sf.api?

Reply:
Just that the `sf.api` namespace seems somewhat redundant with the move to `xcc` as the API client backend 🙂

Reply:
Oh, I see!

Perhaps we can convert `sf.api` from a package to a module with the code for both DeviceSpec and Result.

Reply:
> Perhaps we can convert sf.api from a package to a module

Sounds like a good idea! Either that or just move `result.py` and `device_spec.py` into the root SF folder. If keeping it as one module, what would be the suggested name? `api.py`, `device_spec.py`?

Reply:
Hmm, I think I would go with either `api.py` or `remote.py`.

Reply:
I'm not really sure. Would it perhaps even make more sense to have both a `device_spec.py` _and_ a `result.py` module? The `DeviceSpec` is purely remote, but `Result` is actually used for both local and remote runs.

Reply:
That's a good point! I think you've convinced me to extract them as separate modules.